### PR TITLE
[Tour of Beam] [Task] Fix dependency management for 2.44 Playground java runner

### LINKED ISSUE #25262: [Tour of Beam] [Task] Fix dependency management for 2.44 Playground java runner
### What needs to happen?
Playground java runner is broken for 2.44 Beam due to sdk java harness dependency changes
 

### Issue Priority
Priority: 3

### Components:
tour-of-beam, playground

### PR DESCRIPTION
**Pull request by olehborysevych**
Change dependency management for Playground java runner to use 2.44.
This change was required due to breaking changes in 2.44 with java sdk harness
Also removing lintdocker since it doesn't support valid syntax of redeclaring ARGs

resolves #25262
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by olehborysevych** [comment]
R: @rshamunov @TSultanov

File: playground/backend/containers/java/Dockerfile [hunk 1]
```diff
@@ -57,48 +68,30 @@ COPY --from=build /go/bin/server_java_backend /opt/playground/backend/
...
-    mv junit-$JUNIT_VERSION.jar /opt/apache/beam/jars/junit.jar
-# Install Beam Vendor gRPC
-RUN wget https://repo1.maven.org/maven2/org/apache/beam/beam-vendor-grpc-$GRPC_VERSION/$BEAM_VENDOR_GRPC_VERSION/beam-vendor-grpc-$GRPC_VERSION-$BEAM_VENDOR_GRPC_VERSION.jar &&\
-    mv beam-vendor-grpc-$GRPC_VERSION-$BEAM_VENDOR_GRPC_VERSION.jar /opt/apache/beam/jars/beam-vendor-grpc.jar
-# Install Kafka IO
-RUN wget https://repo1.maven.org/maven2/org/apache/beam/beam-sdks-java-io-kafka/$BEAM_VERSION/beam-sdks-java-io-kafka-$BEAM_VERSION.jar &&\
-    mv beam-sdks-java-io-kafka-$BEAM_VERSION.jar /opt/apache/beam/jars/beam-sdks-java-io-kafka.jar
-# Install Apache Kafka Clients
+# Install kafka client
 RUN wget https://repo1.maven.org/maven2/org/apache/kafka/kafka-clients/$KAFKA_CLIENTS_VERSION/kafka-clients-$KAFKA_CLIENTS_VERSION.jar &&\
```
Comment (TSultanov) [inline playground/backend/containers/java/Dockerfile:80]: Please add comment that we are explicitly downgrading Kafka due to provided version not working with our emulator

File: playground/backend/containers/java/build.gradle [hunk 2]
```diff
@@ -69,9 +69,9 @@ docker {
                   project.docker_image_default_repo_root)
   files "./build/"
   tags containerImageTags()
-  buildArgs(['BASE_IMAGE': project.rootProject.hasProperty(["base-image"]) ?
-                           project.rootProject["base-image"] :
-                           "apache/beam_java8_sdk" ])
+  buildArgs(['BEAM_VERSION': project.rootProject.hasProperty(["sdk-tag"]) ?
+                           project.rootProject["sdk-tag"] :
+                           "2.44.0" ])
```
Comment (TSultanov) [inline playground/backend/containers/java/build.gradle:74]: Can we avoid hard-coding the version?

File: playground/backend/containers/java/Dockerfile (hunk 1 above)
Comment (olehborysevych) [reply playground/backend/containers/java/Dockerfile:80]: Done

File: playground/backend/containers/java/build.gradle (hunk 2 above)
Comment (olehborysevych) [reply playground/backend/containers/java/build.gradle:74]: I didn't find good way to do this. Our approach for java runner is different from other runners that we can only use release beam version

**Comment by olehborysevych** [comment]
R: @damccorm

File: playground/backend/containers/java/build.gradle (hunk 2 above)
Comment (damccorm) [reply playground/backend/containers/java/build.gradle:74]: Could we at least pull it out as a constant at the top of the file?

**Review by damccorm — APPROVED** [review]
LGTM

**Pull request merged**